core/pkg/sync-store/sync_store.go

```diff
@@ -0,0 +1,235 @@
+package sync_store
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"sync"
+	"time"
+
+	"github.com/open-feature/flagd/core/pkg/logger"
+	isync "github.com/open-feature/flagd/core/pkg/sync"
+	"github.com/open-feature/flagd/core/pkg/sync/file"
+	"github.com/open-feature/flagd/core/pkg/sync/kubernetes"
+	"go.uber.org/zap"
+)
+
+var (
+	regCrd  *regexp.Regexp
+	regFile *regexp.Regexp
+)
+
+func init() {
+	regCrd = regexp.MustCompile("^core.openfeature.dev/")
+	regFile = regexp.MustCompile("^file:")
+}
+
+type SyncStore struct {
+	ctx          context.Context
+	syncHandlers map[string]*syncHandler
+	logger       *logger.Logger
+	mu           *sync.Mutex
+	SyncBuilder  SyncBuilderInterface
+}
+
+type syncHandler struct {
+	subs       map[interface{}]storedChannels
+	dataSync   chan isync.DataSync
+	cancelFunc context.CancelFunc
+	syncRef    isync.ISync
+}
+
+type storedChannels struct {
+	errChan  chan error
+	dataSync chan isync.DataSync
+}
+
+func NewSyncStore(ctx context.Context, logger *logger.Logger) SyncStore {
+	return SyncStore{
+		ctx:          ctx,
+		syncHandlers: map[string]*syncHandler{},
+		logger:       logger,
+		mu:           &sync.Mutex{},
+		SyncBuilder:  &SyncBuilder{},
+	}
+}
+
+func (s *SyncStore) FetchAllFlags(ctx context.Context, key interface{}, target string) (isync.DataSync, error) {
+	dataSyncChan := make(chan isync.DataSync, 1)
+	errChan := make(chan error, 1)
+	s.mu.Lock()
+	_, ok := s.syncHandlers[target]
```

Comment:
I'd assign the syncHandler to a var so the Unlock can be called directly after. Also this can be RLock/RUnlock unless I've misunderstood